### PR TITLE
Only return state events that the AS passed in via `state_events_at_start` (MSC2716)

### PR DESCRIPTION
**Pull request by MadLittleMods**
Update `/batch_send` endpoint to only return `state_events` created by the `state_events_from_before` passed in.

As discovered by @Half-Shot in
https://github.com/matrix-org/matrix-doc/pull/2716#discussion_r684158448

Part of [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716)

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [ ] ~~Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)~~
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: synapse/rest/client/v1/room.py [hunk 1]
```diff
@@ -438,6 +438,7 @@ async def on_POST(self, request, room_id):
         prev_state_ids = list(prev_state_map.values())
```
Comment (erikjohnston) [inline synapse/rest/client/v1/room.py:438]: I've just realised that we should probably not be adding this class to `v1/room.py`, but instead somewhere in `v2/`. If only someone had reviewed the PR that added this endpoint.................. :facepalm:

**Review by erikjohnston — APPROVED** [review]

File: synapse/rest/client/v1/room.py (hunk 1 above)
Comment (MadLittleMods) [reply synapse/rest/client/v1/room.py:438]: I can switch it over in another PR 😁

**Comment by MadLittleMods** [comment]
@erikjohnston Thanks for the review and noticing the endpoint being organized in the wrong area 🦑

File: synapse/rest/client/v1/room.py (hunk 1 above)
Comment (MadLittleMods) [reply synapse/rest/client/v1/room.py:438]: https://github.com/matrix-org/synapse/pull/10576 ⏩

**Pull request merged**